src/shader_recompiler/frontend/translate/vector_alu.cpp

```diff
@@ -294,6 +314,20 @@ void Translator::V_SUBREV_I32(const GcnInst& inst) {
     // TODO: Carry-out
 }
 
+void Translator::V_MAD_U64_U32(const GcnInst& inst) {
+
+    const auto src0 = GetSrc<IR::U32>(inst.src[0]);
+    const auto src1 = GetSrc<IR::U32>(inst.src[1]);
+    const auto src2 = GetSrc64<IR::U64>(inst.src[2]);
+
+    IR::U64 result;
+    result = ir.IMul(src0, src1);
+    result = ir.IAdd(ir.UConvert(64, result), src2);
+
+    const IR::VectorReg dst_reg{inst.dst[0].code};
+    ir.SetVectorReg64(dst_reg, result);
```

Comment:
Docs say this also sets VCC but it is unclear to me if it is full 64bit register or only lower 32bits